static/app/views/insights/common/components/modulePageProviders.tsx

```diff
@@ -30,19 +31,42 @@ export function ModulePageProviders({moduleName, pageTitle, children, features}:
     .filter(Boolean)
     .join(' — ');
 
+  const defaultBody = (
+    <Layout.Page>
+      <Feature features={features} organization={organization} renderDisabled={NoAccess}>
+        <NoProjectMessage organization={organization}>{children}</NoProjectMessage>
+      </Feature>
+    </Layout.Page>
+  );
+
   return (
     <PageFiltersContainer>
       <SentryDocumentTitle title={fullPageTitle} orgSlug={organization.slug}>
-        <Layout.Page>
-          <Feature
-            features={features}
-            organization={organization}
-            renderDisabled={NoAccess}
-          >
-            <NoProjectMessage organization={organization}>{children}</NoProjectMessage>
-          </Feature>
-        </Layout.Page>
+        <UpsellPageHook id={sidebarIdMap[moduleName]}>{() => defaultBody}</UpsellPageHook>
       </SentryDocumentTitle>
     </PageFiltersContainer>
   );
 }
+
+// This matches ids in the sidebar items and in the hook in getsentry
+export const sidebarIdMap: Record<TitleableModuleNames, string> = {
+  ai: 'llm-monitoring',
+  'mobile-ui': 'performance-mobile-ui',
+  cache: 'performance-cache',
+  db: 'performance-database',
+  http: 'performance-http',
+  resource: 'performance-browser-resources',
+  screen_load: 'performance-mobile-screens',
+  app_start: 'performance-mobile-app-startup',
+  vital: 'performance-webvitals',
+  queue: 'performance-queues',
+};
```

Comment:
🤔 since these are full page upsells, do you need to bother with this mapping and lookup? You don't need to pass these a sidebar ID, you can just pass through the module name, no?

Reply:
You're right, I could just pass in a module name, and that would probably be my preference. 
But to share logic with the sidebar item hook, we'll have to map ids to module names (or vise versa) somewhere. 

Perhaps I do the mapping in getsentry and pass in module name here for a better api?

Reply:
I've updated to accept moduleName instead!